From 5acf1d9f19895368d8ad956a88d0290f8db23c2f Mon Sep 17 00:00:00 2001
From: Bert van Hall <bert.vanhall@avionic-design.de>
Date: Wed, 21 Oct 2015 15:45:06 +0200
Subject: [PATCH 1/3] Compatibility with pre 3.6 kernels

Expand extended_frontend.h a bit to ensure pre 3.6 Linux kernel
compatibility.

Signed-off-by: Bert van Hall <bert.vanhall@avionic-design.de>
---
 extended_frontend.h | 49 +++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 49 insertions(+)

diff --git a/extended_frontend.h b/extended_frontend.h
index af393db..26005f2 100644
--- a/extended_frontend.h
+++ b/extended_frontend.h
@@ -76,6 +76,15 @@ typedef enum fe_alpha {
 } fe_alpha_t;
 #endif
 
+#if DVB_API < 57
+enum fe_interleaving {
+	INTERLEAVING_NONE,
+	INTERLEAVING_AUTO,
+	INTERLEAVING_240,
+	INTERLEAVING_720,
+};
+#endif
+
 typedef enum fe_siso_miso {
         SISO,
         MISO,
@@ -99,10 +108,50 @@ typedef enum fe_siso_miso {
 #endif
 #endif                                         //// \END     REMOVE UP TO HERE!!!
 
+#ifndef SYS_DVBC_ANNEX_A
+#define SYS_DVBC_ANNEX_A SYS_DVBC_ANNEX_AC
+#endif
+
 #ifndef SYS_TURBO  // remove later.
 #define SYS_TURBO (SYS_DVBT2 + 1) /* correct in any case. */
 #endif
 
+#ifndef SYS_DVBC_ANNEX_C
+#define SYS_DVBC_ANNEX_C (SYS_TURBO + 1)
+#endif
+
+#ifndef SYS_DTMB
+#define SYS_DTMB SYS_DMBTH
+#endif
+
+#ifndef FEC_2_5
+#define FEC_2_5 (FEC_9_10 + 1)
+#endif
+
+#ifndef QAM_4_NR
+#define QAM_4_NR (DQPSK + 1)
+#endif
+
+#ifndef TRANSMISSION_MODE_C1
+#define TRANSMISSION_MODE_C1 (TRANSMISSION_MODE_32K + 1)
+#endif
+
+#ifndef TRANSMISSION_MODE_C3780
+#define TRANSMISSION_MODE_C3780 (TRANSMISSION_MODE_32K + 2)
+#endif
+
+#ifndef GUARD_INTERVAL_PN420
+#define GUARD_INTERVAL_PN420 (GUARD_INTERVAL_19_256 + 1)
+#endif
+
+#ifndef GUARD_INTERVAL_PN595
+#define GUARD_INTERVAL_PN595 (GUARD_INTERVAL_19_256 + 2)
+#endif
+
+#ifndef GUARD_INTERVAL_PN945
+#define GUARD_INTERVAL_PN945 (GUARD_INTERVAL_19_256 + 3)
+#endif
+
 #ifndef DTV_ENUM_DELSYS // remove later.
 #define DTV_ENUM_DELSYS 44
 #endif
-- 
2.1.0

